utils/core/src/main/kotlin/EnvVarAuthenticator.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2020-2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.core
+
+import java.net.Authenticator
+import java.net.PasswordAuthentication
+
+import org.ossreviewtoolkit.utils.common.Os
+
+/**
+ * A simple non-caching authenticator that reads credentials from [environment variables][env].
```

Comment:
Suggestion: Maybe name the variables explicitly in the comment, as they are hard-coded.